clang/lib/Driver/Driver.cpp

```diff
@@ -3036,7 +3105,12 @@ class OffloadingActionBuilder final {
         for (Action *&A : OpenMPDeviceActions) {
           assert(isa<CompileJobAction>(A));
           OffloadAction::DeviceDependences DDep;
-          DDep.add(*A, **TC, /*BoundArch=*/nullptr, Action::OFK_OpenMP);
+          if (GpuArchList.size())
```

Comment:
i think this is about compiling for multiple gpu arch at the same time. if trunk doesn't do that for nvptx, we don't need to do it for amdgcn as part of the minimum viable patch